Add GHDL waveform documentation (#1980)
cocotb/cocotb: documentation/source/simulator_support.rst

Noteworthy is that despite GHDL being a VHDL simulator, it implements the VPI interface.

.. _sim-ghdl-waveforms:

Waveforms
---------

To get waveforms in VCD format, set the :make:var:`SIM_ARGS` option to ``--vcd=anyname.vcd``,
for example in a Makefile:

.. code-block:: make

    SIM_ARGS+=--vcd=anyname.vcd

The option can be set on the command line, as shown in the following example.

.. code-block:: bash

    SIM_ARGS=--vcd=anyname.vhd make SIM=ghdl

A VCD file named ``anyname.vcd`` will be generated in the current directory.

> :make:var:`SIM_ARGS` can also be used to pass command line arguments related to other waveform formats supported by GHDL like :ref:`Fast Signal Trace (FST) <ghdl:export_waves>`, which is compressed; or :ref:`GHDL Wave File (GHW) <ghdl:export_waves>`, which supports any VHDL type. 

Reviewer: Since these links all go to the same place, I'm tempted to cut this back to:
```suggestion
:make:var:`SIM_ARGS` can also be used to pass command line arguments related to :ref:`other waveform formats supported by GHDL <ghdl:export_waves>`.

Reviewer 2: LGTM. I think that mentioning FST and GHW would be desirable because those are less known but significantly more efficient formats than VCD. Anyway, I understand that, in the context of cocotb, that might not be so relevant. Hence, I believe both wordings are acceptable.

Noteworthy is that despite GHDL being a VHDL simulator, it implements the VPI interface.

.. _sim-ghdl-waveforms:

Waveforms
---------

To get waveforms in VCD format, set the :make:var:`SIM_ARGS` option to ``--vcd=anyname.vcd``,
for example in a Makefile:

.. code-block:: make

    SIM_ARGS+=--vcd=anyname.vcd

The option can be set on the command line, as shown in the following example.

.. code-block:: bash

    SIM_ARGS=--vcd=anyname.vhd make SIM=ghdl

A VCD file named ``anyname.vcd`` will be generated in the current directory.

The :make:var:`SIM_ARGS` option allows access to all

> [GHDL Command Line Run (-r) options](https://ghdl.readthedocs.io/en/latest/using/Simulation.html#using-simulation).

Reviewer 2: Alternatively, you can add an intersphinx link (see https://github.com/ghdl/ghdl-cosim/blob/master/doc/conf.py#L132) and replace these references with:

```rst
:ghdl:`USING:Simulation` or :ghdl:`simulation_options`
:ghdl:`export_waves`
```

The syntax you are using here seems to be Markdown, although the document is written in reStructuredText.

Author: I changed to intersphinx links and correct FST/GHW case.

Reviewer 2: Although CI is not complaining explicitly about it, I'd say you need to add the intersphinx mapping in https://github.com/jwrr/cocotb/blob/jwrr/ghdlwaves/documentation/source/conf.py#L49. Just copy the one I referenced above.

See https://cocotb--1980.org.readthedocs.build/en/1980/simulator_support.html?highlight=ghdl#sim-ghdl-waveforms and:

```
https://readthedocs.org/projects/cocotb/builds/11449536/

/home/docs/checkouts/readthedocs.org/user_builds/cocotb/checkouts/1980/documentation/source/simulator_support.rst:279: WARNING: Unknown interpreted text role "ghdl".
/home/docs/checkouts/readthedocs.org/user_builds/cocotb/checkouts/1980/documentation/source/simulator_support.rst:279: WARNING: Unknown interpreted text role "ghdl".
```

Author: I added ghdl to intersphinx_mapping in conf.py and got the links to use it after a few tries.

Reviewer 2: Great! Thanks for your responsiveness. I know it is annoying at first to do so many iterations for an apparently simple contribution. However, as said, I think that tightly cross-referencing docs of different projects is a good excuse to build a community, given that most of the projects use Sphinx. Hence, I'm really glad that you used intersphinx.

Author: @umarcor thanks for walking me through the process.  Please comment on @eric-wieser latest suggestion that is brief and clear but loses some information about FST and GHW.  I think brief is probably better because clicking the link will provide all extra info.

Noteworthy is that despite GHDL being a VHDL simulator, it implements the VPI interface.

.. _sim-ghdl-waveforms:

Waveforms
---------

To get waveforms in VCD format, set the :make:var:`SIM_ARGS` option to ``--vcd=anyname.vcd``,
for example in a Makefile:

.. code-block:: make

    SIM_ARGS+=--vcd=anyname.vcd

The option can be set on the command line, as shown in the following example.

.. code-block:: bash

    SIM_ARGS=--vcd=anyname.vhd make SIM=ghdl

A VCD file named ``anyname.vcd`` will be generated in the current directory.

> The :make:var:`SIM_ARGS` option allows access to all GHDL Command Line Run (-r)
> options listed in :ref:`ghdl:simulation_options`. This includes other waveform
> generation options including the compressed Fast Signal Trace (FST) format and
> the GHDL Wave File (GHW) which supports any VHDL type. For more information
> see :ref:`ghdl:export_waves`.

Reviewer 3: I would reword this slightly. The first sentence is kind of redundant, and I would use better name for the cross-refs. Riff on this if you want.

```suggestion
:make:var:`SIM_ARGS` can also be used to pass command line arguments related to other waveform formats supported by GHDL like :ref:`Fast Signal Trace (FST) <ghdl:export_waves>`, which is compressed; or :ref:`GHDL Wave File (GHW) <ghdl:export_waves>`, which supports any VHDL type. 
```

Noteworthy is that despite GHDL being a VHDL simulator, it implements the VPI interface.

.. _sim-ghdl-waveforms:

Waveforms
---------

> To get waveforms in VCD format, set the :make:var:`SIM_ARGS` option to ``--vcd=anyname.vcd``,

Reviewer 2: See https://github.com/cocotb/cocotb/pull/1972#discussion_r453325569.

Author: I pushed a brief mention about other GHDL waveform formats

for example in a Makefile:

>   .. code-block:: make

Reviewer: ```suggestion
.. code-block:: make
```

    SIM_ARGS+=--vcd=anyname.vcd

The option can be set on the command line, as shown in the following example.

  .. code-block:: bash

    make SIM=ghdl SIM_ARGS=--vcd=anyname.vcd

    # or define the SIM_ARGS environment variable in bash

    export SIM_ARGS=--vcd=anyname.vhd; make SIM=ghdl

    # or in csh

>     setenv SIM_ARGS=--vcd=anyname.vhd; make SIM=ghdl

Reviewer 3: I want to get some other opinions on this. It does work and it's convenient, but I feel like recommending this will lead to people forgetting or not realizing it's set and it causing them issues. So I'd rather not mention it.

Author: I accepted @eric-wieser suggestion of `SIM_ARGS=--vcd=anyname.vhd make SIM=ghdl`

> The option can be set on the command line, as shown in the following example.
> 
>   .. code-block:: bash
> 
>     make SIM=ghdl SIM_ARGS=--vcd=anyname.vcd
> 
>     # or define the SIM_ARGS environment variable in bash
> 
>     export SIM_ARGS=--vcd=anyname.vhd; make SIM=ghdl
> 
>     # or in csh
> 
>     setenv SIM_ARGS=--vcd=anyname.vhd; make SIM=ghdl

Reviewer: Didn't we decide that we want to recommend none of these, but instead:
```suggestion
The option can be set on the command line, as shown in the following example.

.. code-block:: bash

    SIM_ARGS=--vcd=anyname.vhd make SIM=ghdl
```

Reviewer: Frankly though, I don't think it's the job of `simulator_support.rst` to tell users how to set make variables - the page listing make variables should cover that.
